### PR TITLE
sql: write to the event log table when changing privileges

### PR DESCRIPTION
**Pull request by neeral**
Helps towards https://github.com/cockroachdb/cockroach/issues/13492.

Release note (admin ui change): Changing privileges (i.e. grants or
revokes) now causes an avent to be logged and displayed in the admin ui

cc: @jordanlewis could you please review this or suggest someone who would be suitable?

### CODE REVIEW

**Review by knz — APPROVED** [review]
@thtruo you'll like this

> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55612#-:-MJkX-Yp4RJ7MeuRDQuF:b-nl8q8o)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Comment by knz** [comment]
Thanks neeral

**Comment by knz** [comment]
I think there should be a review from the SQL experience team as well though.

**Review by arulajmani — CHANGES_REQUESTED** [review]
Thanks for your contribution @neeral. I left a small suggested change, and should be good to go once made. :) 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55612#-:-MJmLQUF4v2jEwgDy3LM:baux2g2)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @neeral)

___
*[pkg/sql/grant_revoke.go, line 264 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55612#-MJmJ0Rl8WkNauHjRvEp:-MJmJ0Rl8WkNauHjRvEq:b-pyuael) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7e3892a39a1a84fbae96fecc8818ef0ecdbe975/pkg/sql/grant_revoke.go#L264)):*
> ```Go
> 		struct {
> 			Target     string
> 			User       string
> ```

Looks like for all the other event logs, `User` refers to the user performing the action. Can we add a `Grantees` field to the `EventInfo` struct instead of re-using `User` to represent all the users that the privilege(s) has been granted/revoked from? 

This will also allow us to then capture the user who performed the operation and thereby surface that information in the event log appropriately! 
___
*[pkg/ui/src/util/events.ts, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55612#-MJmKk7aAe9fFQrz8BsC:-MJmKk7aAe9fFQrz8BsD:b-4h9x1m) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7e3892a39a1a84fbae96fecc8818ef0ecdbe975/pkg/ui/src/util/events.ts#L87)):*
> ```XML
>       return `Table statistics refreshed for ${info.TableName}`;
>     case eventTypes.GRANT_PRIVILEGE:
>       return `Privileges granted: ${info.Privileges} added to User(s) ${info.User} on ${info.Target}`;
> ```

Once we make the suggested change above, we can reword this to something like:
`Privileges granted: User ${info.User} granted ${info.Privileges} to ${info.Grantees} on ${info.Target}`

**Review by neeral — COMMENTED** [review]
Hi @arulajmani thanks for the feedback. I've updated the PR after addressing your comments.

PTAL.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55612#-:-MJnagx9Cp4GKBsN_c0E:b-4on1cb)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani and @knz)

___
*[pkg/sql/grant_revoke.go, line 264 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55612#-MJmJ0Rl8WkNauHjRvEp:-MJnadNz7xxsLghQoY0y:b-cfj7zj) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7e3892a39a1a84fbae96fecc8818ef0ecdbe975/pkg/sql/grant_revoke.go#L264)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Looks like for all the other event logs, `User` refers to the user performing the action. Can we add a `Grantees` field to the `EventInfo` struct instead of re-using `User` to represent all the users that the privilege(s) has been granted/revoked from? 

This will also allow us to then capture the user who performed the operation and thereby surface that information in the event log appropriately! 
</blockquote></details>

Thanks for the feedback.
___
*[pkg/ui/src/util/events.ts, line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55612#-MJmKk7aAe9fFQrz8BsC:-MJnaS_mEiVHdJG2iiAl:b78mc1a) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7e3892a39a1a84fbae96fecc8818ef0ecdbe975/pkg/ui/src/util/events.ts#L87)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Once we make the suggested change above, we can reword this to something like:
`Privileges granted: User ${info.User} granted ${info.Privileges} to ${info.Grantees} on ${info.Target}`
</blockquote></details>

OK. Addressed.

**Review by arulajmani — APPROVED** [review]
Thanks for making the changes, this is :lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55612#-:-MK0--bIDy-nMdpSKUoY:b8xzjac)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @arulajmani and @knz)

**Comment by knz** [comment]
nice :100: 

bors r=knz,arulajmani

**Comment by knz** [comment]
@piyush-singh @thtruo do we want to backport this?

**Comment by piyush-singh** [comment]
I think it would be worth getting this into a 20.2 point release; I know a few customers have requested this in the past. This is probably more important now that we have introduced several new more granular privileges in 20.2.

**Comment by arulajmani** [comment]
Just 20.2 or do we want this in 20.1 as well?

**Comment by piyush-singh** [comment]
If it isn't too much trouble, 20.1 would be great as well.

**Comment by Amruta-Ranade** [comment]
@neeral Thank you for contributing to CockroachDB this year. As a token of our appreciation, we would like to send you a gift. Please DM me on our [community Slack](https://cockroa.ch/slack) @Amruta so I can send you a link. (If you don’t want a gift, we also have a charitable contribution choice.) All orders need to be in by December 13, so please contact me as soon as possible :)
